### ISSUE TITLE
Is there an input-queued linklayer interface in INET?

### ISSUE THREAD

**Issue by kalsasdf** [inet-framework/inet#864]
Hi, I notice that the current linklayer interfaces in **INET** are all in the output queue model, that is, packets are queued at the output port. Take the EthernetMacBase mdule in EthernetInterface.ned as an example, 
<img width="375" alt="1678298339725" src="https://user-images.githubusercontent.com/31431263/223792761-77a3116d-a953-49c0-8b7d-f8a4e3dbb717.png">
it needs to bind a queue as its buffer **txQueue** in the initialize() function using the getQueue() function.
<img width="299" alt="1678298479893" src="https://user-images.githubusercontent.com/31431263/223793265-3f53ed8a-da8d-4d09-b90c-8a653da9d8c9.png">
This is fine for host nodes, but for switches, we know that there are more input-queued switches in reality, that is, packets are queued at the input port of the switch. And there are many protocols that rely on this feature, such as the flow control mechanism PFC (priority flow control) in data centers. I would like to know if INET has an input-queued linklayer interface or if there will be one in the future?

**Comment by levy** [comment]
No, currently there's no such thing in INET, and we do not have this feature on our roadmap. The queues are always part of the network interface, and packets are always enqueued in the outgoing network interface after the forwarding decision has been made.

I'm don't exactly know how this feature should work, so I'm not sure how it could be implemented. Maybe it's easier to implement this than it sounds for the first time. INET is already very modular and flexible now with respect to the network node architecture, so maybe very little C++ programming is required.

So if you could give us more details that would be great!

**Comment by levy** [comment]
I have just pushed two example configurations in the queueing tutorial in 0f2464e4551394702e236a1035ab9ca19468c3b5 that demonstrates a possible solution for the different queueing architectures (see InputQueueSwitching and OutputQueueSwitching networks and configuration). For real network nodes the network architecture would be more complicated because the switching mechanism consists of several additional modules including MessageDispatchers, etc.

This comment is more like a reminder, not a real solution.
